src/main/java/seedu/address/logic/commands/eventvolunteercommands/EventAddVolunteerCommand.java

```diff
@@ -63,11 +68,40 @@ public CommandResult execute(Model model) throws CommandException {
         if (eventToAssign.hasVolunteer(volunteerToAssign)) {
             throw new CommandException(MESSAGE_DUPLICATE_VOLUNTEER);
         }
+
+        Set<EventName> assignedEvents = volunteerToAssign.getAssignedEvents();
+        for (EventName eventName : assignedEvents) {
+            Predicate<Event> eventPredicate = e -> e.getEventName().equals(eventName);
+            Event otherEvent = model.getEventStorage().getEvent(eventName);
+            if (hasClashingEvents(eventToAssign, otherEvent)) {
+                throw new CommandException(MESSAGE_CLASHING_EVENTS);
+            }
+        }
         volunteerToAssign.addEvent(eventToAssign);
         eventToAssign.addVolunteer(volunteerToAssign);
         return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(eventToAssign),
                 eventToAssign.getAssignedVolunteers().size()));
     }
+
+    /**
+     * Checks if {@code eventToAssign} clashes with {@code otherEvent}
+     * @param eventToAssign The Event to be assigned.
+     * @param otherEvent The Event to be checked.
+     * @return true if the two events clash.
+     */
+    public boolean hasClashingEvents(Event eventToAssign, Event otherEvent) {
```

Comment:
might want to draw a diagram to confirm the logic of this. i'm pretty sure there are counterexamples such as same start time but different end times. since `isAfter()` and `isBefore()` returns *false* for equal LocalDateTime objects. 

- to capture the same time or after effect you can do: `!date1.isBefore(date2);`
- to capture the same time or before effect you can do: `!date1.isAfter(date2);`

Reply:
okie